tests/Backend/CommonPart1/CreateTableWithConfigurationTest.php

```diff
@@ -0,0 +1,344 @@
+                        'description' => 'first ever',
+                    ],
+                ],
+            ])
+            ->setDescription('Configuration for Custom Queries');
+        $this->componentsClient->addConfiguration($configuration);
+
+        // create table from config
+        $tableName = 'custom-table-1';
+        $configurationOptions = (new TableWithConfigurationOptions())
+            ->setTablename($tableName)
+            ->setConfigurationId($configurationId);
+        $tableId = $this->_client->createTableWithConfiguration(
+            $this->getTestBucketId(),
+            $configurationOptions
+        );
+
+        $table = $this->_client->getTable($tableId);
+        $this->assertArrayHasKey('displayName', $table['bucket']);
+
+        $this->assertEquals($tableId, $table['id']);
+        $this->assertEquals($tableName, $table['name']);
+        $this->assertEquals($tableName, $table['displayName'], 'display name is same as name');
+        $this->assertNotEmpty($table['created']);
+        $this->assertNotEquals('0000-00-00 00:00:00', $table['created']);
+    }
+
+    public function testTableCreateWithMeaningFullQueryAsSecond(): void
+    {
+        $configurationId = 'main-1';
+
+        // create test configuration
+        $configuration = (new Configuration())
+            ->setComponentId(self::COMPONENT_ID)
+            ->setConfigurationId($configurationId)
+            ->setName('Main')
+            ->setState(['stateValue' => 'some-value'])
+            ->setConfiguration([
+                'migrations' => [
+                    [
+                        'sql' => 'SELECT 1',
+                        'description' => 'first ever',
+                    ],
+                    [
+                        'sql' => 'CREATE TABLE {{ id(bucketName) }}.{{ id(tableName) }} (id integer, name varchar(100))',
+                        'description' => 'first ever',
+                    ],
+                ],
+            ])
+            ->setDescription('Configuration for Custom Queries');
+        $this->componentsClient->addConfiguration($configuration);
+
+        // create table from config
+        $tableName = 'custom-table-1';
+        $configurationOptions = (new TableWithConfigurationOptions())
+            ->setTablename($tableName)
+            ->setConfigurationId($configurationId);
+        $tableId = $this->_client->createTableWithConfiguration(
+            $this->getTestBucketId(),
+            $configurationOptions
+        );
+
+        $table = $this->_client->getTable($tableId);
+        $this->assertArrayHasKey('displayName', $table['bucket']);
+
+        $this->assertEquals($tableId, $table['id']);
+        $this->assertEquals($tableName, $table['name']);
+        $this->assertEquals($tableName, $table['displayName'], 'display name is same as name');
+        $this->assertNotEmpty($table['created']);
+        $this->assertNotEquals('0000-00-00 00:00:00', $table['created']);
```

Comment:
Tady by se mělo validovat, že je nastavené lastMigrationId možná? Asi to nemáme v response, ale možná bysme měli

Reply:
plus zas metadata